### PR TITLE
add deadline filter in main menu

### PR DESCRIPTION
**Pull request by Voxanimus**

### CODE REVIEW

File: app/src/main/java/com/github/multimatum_team/multimatum/activity/MainActivity.kt [hunk 1]
```diff
@@ -38,28 +39,68 @@ class MainActivity : AppCompatActivity() {
 
     private val deadlineListViewModel: DeadlineListViewModel by viewModels()
 
+    val filters = listOf("All", "Mine", "Groups")
```
[the remark this answers is not in the document]
Comment (Voxanimus) [reply app/src/main/java/com/github/multimatum_team/multimatum/activity/MainActivity.kt:42]: That's a nice idea, I think the best way to implements this is to keep the All and the MIne filter, and make this list dynamic with all group name in wich the user is.

File: app/src/main/java/com/github/multimatum_team/multimatum/adaptater/DeadlineAdapter.kt [hunk 2]
```diff
@@ -70,11 +81,48 @@ class DeadlineAdapter(
     private val inflater: LayoutInflater =
         context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
 
-    fun setDeadlines(deadlines: Map<DeadlineID, Deadline>) {
-        dataSource = sortDeadline(deadlines)
+    fun setDeadlines(deadlines: Map<DeadlineID, Deadline>, filter: String = "") {
```
Comment (felko) [inline app/src/main/java/com/github/multimatum_team/multimatum/adaptater/DeadlineAdapter.kt:84]: I don't think it's a good idea to represent the filter as a string

File: app/src/main/java/com/github/multimatum_team/multimatum/adaptater/DeadlineAdapter.kt [hunk 3]
```diff
@@ -70,11 +81,48 @@ class DeadlineAdapter(
...
+        dataSource = sortDeadline(deadlines, filter)
         notifyDataSetChanged()
     }
 
+    private fun filterByGroup(deadlines: MutableList<Pair<DeadlineID, Deadline>>, filter: String):
+            MutableList<Pair<DeadlineID, Deadline>>{
+        if (filter.isBlank() or filter.isEmpty()) throw IllegalArgumentException()
```
Comment (felko) [inline app/src/main/java/com/github/multimatum_team/multimatum/adaptater/DeadlineAdapter.kt:91]: You woudn't need this check if the filter was represented by a `DeadlineOwner`

**Review by felko — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/multimatum_team/multimatum/adaptater/DeadlineAdapter.kt [hunk 4]
```diff
@@ -34,31 +39,37 @@ Class which is used to show the deadline in a clear list.
...
+     * state to filter deadlines not in groups
+     */
+    MINE,
+
+    /**
+     * don't filter deadlines
+     */
+
+    ALL
+}
```
Comment (macintosh1398) [inline app/src/main/java/com/github/multimatum_team/multimatum/adaptater/DeadlineAdapter.kt:58]: It is a detail, but for readability purposes, I suggest to put these comments as "line comments " and reduce the space between elements.

**Review by macintosh1398 — APPROVED** [review]
Good addition of the spinner. The code looks good, but just look at my comment.

**Pull request merged**